### PR TITLE
CDAP-13152 Handle null input schema

### PR DESCRIPTION
**Pull request by anwar6953**
[CDAP-13152](https://issues.cask.co/browse/CDAP-13152)

See the JIRA for the stack trace / error that is encountered when input schema for the plugin is not configured.
The [getInputSchema](https://github.com/caskdata/cdap/blob/release/4.3/cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/StageContext.java#L123) method return type is `@Nullable`, so we need to be able to handle it.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -94,6 +94,13 @@
     </repository>
   </repositories>
 
+  <pluginRepositories>
```
Comment (albertshau) [inline pom.xml]: was this added on purpose? don't think we should be depending on any snapshot dependencies.

Comment (albertshau) [reply pom.xml]: just read the commit message, I think 1.0.1 is published.

File: src/main/java/co/cask/hydrator/plugin/PartitionedFileSetSinkConfig.java [hunk 2]
```diff
@@ -140,7 +140,10 @@ protected Partitioning getPartitioningWithoutMacro(Schema inputSchema) {
   /**
    * Validate plugin configuration at deployment stage.
    */
-  protected void validate(Schema inputSchema) {
+  protected void validate(@Nullable Schema inputSchema) {
+    if (inputSchema == null) {
+      return;
```
Comment (ecfm) [inline src/main/java/co/cask/hydrator/plugin/PartitionedFileSetSinkConfig.java:146]: if we allow `inputSchema` to be null, `getPartitioningWithoutMacro` method should also be changed, where it assumes `inputSchema` is not null

File: pom.xml (hunk 1 above)
Comment (anwar6953) [reply pom.xml]: Hmm, doesn't seem so:
` Could not find artifact co.cask:cdap-maven-plugin:jar:1.0.1`

File: src/main/java/co/cask/hydrator/plugin/PartitionedFileSetSinkConfig.java (hunk 2 above)
Comment (anwar6953) [reply src/main/java/co/cask/hydrator/plugin/PartitionedFileSetSinkConfig.java:146]: Good point. Yea, I'll need to update that method also.

**Comment by ecfm** [comment]
LGTM

File: pom.xml (hunk 1 above)
Comment (albertshau) [reply pom.xml]: hm 1.0.0 is there. I would check with @sreevatsanraman with what the difference in 1.0.1-SNAPSHOT is, we really shouldn't have any snapshot version anywhere.

**Comment by anwar6953** [comment]
Also bumped the project version to 2.0.2.

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by anwar6953** [comment]
Squashed commits, merging.

**Pull request merged**